src/inference/dev_api/openvino/runtime/threading/istreams_executor.hpp

```diff
@@ -133,6 +134,7 @@ class OPENVINO_RUNTIME_API IStreamsExecutor : virtual public ITaskExecutor {
                ov::hint::SchedulingCoreType thread_preferred_core_type = ov::hint::SchedulingCoreType::ANY_CORE,
                bool cpu_reservation = false,
                bool cpu_pinning = false,
+               bool cores_limit = true,
```

Comment:
The ctor becomes too complicates: hard to understand which combinations are really makes sence (or at least valid) and which not. So the propose for the future refactoring is to hide the complexity by adding several simple ctors or even introduce some modes.

Reply:
> The ctor becomes too complicates: hard to understand which combinations are really makes sence (or at least valid) and which not. So the propose for the future refactoring is to hide the complexity by adding several simple ctors or even introduce some modes.

Will address this issue in future refactoring.